dcpy/lifecycle/ingest/configure.py

```diff
@@ -113,11 +115,18 @@ def get_config(dataset: str, version: str | None = None) -> Config:
     processing_steps = template.processing_steps
 
     if template.target_crs:
-        reprojection = FunctionCall(
+        reprojection = PreprocessingStep(
             name="reproject", args={"target_crs": template.target_crs}
         )
         processing_steps = [reprojection] + processing_steps
 
+    if mode:
+        modes = {s.mode for s in processing_steps}
+        if mode not in modes:
```

Comment:
worth noting here that filtering processing steps happen at time of config generation. Meaning, the config.json that gets dumped with a dataset ONLY has the steps that were actually run.

writing this now, it might be worth keeping `mode` as a saved field in the config? In that case, we could include all preprocessing steps (and it can be infered by `mode` which one was run), though I sort of like the fact that steps are only included if they were run. But then it doesn't quite reflect the definition in the original template anymore. @alexrichey would be curious if you have any strong feelings

Reply:
@fvankrieken Including all the fields + mode seems sensible. Or you could add a flag called `skipped`, or something like, that to each step that was omitted?